client/components/DataManagement/hooks.js

```diff
@@ -0,0 +1,232 @@
+import { useMemo, useState } from 'react';
```

Comment:
Great idea splitting out the logic into hooks. I definitely appreciate that idea.

I was thinking maybe can rename the file to refer to the idea of filtering and sorting instead of the more general "hooks.js"? 

If we end up another complex feature like this where it would be too complicated to dump it into the component, in that case it seems like we would want to create a separate file instead of intermingling it in this hooks file.

Reply:
Agreed! I opted for renaming the file for now. A more common solution if hooks become more prevalent in the future would be a hooks folder in the component folder with a separate file for each hook.